src/Analyzers/MSTest.Analyzers/DynamicDataShouldBeValidAnalyzer.cs

```diff
@@ -276,8 +276,7 @@ private static void AnalyzeDataSource(SymbolAnalysisContext context, AttributeDa
         // Validate member return type.
         ITypeSymbol? memberTypeSymbol = member.GetMemberType();
         if (memberTypeSymbol is INamedTypeSymbol memberNamedType
-            && (!SymbolEqualityComparer.Default.Equals(memberNamedType.ConstructedFrom, ienumerableTypeSymbol)
-                || memberNamedType.TypeArguments.Length != 1))
+            && !memberNamedType.OriginalDefinition.Inherits(ienumerableTypeSymbol))
```

Comment:
This is referring to `IEnumerable<T>`, but I think it should refer to `IEnumerable`?

nit: `SpecialType.System_Collections_IEnumerable` can be used.